### ISSUE TITLE
boolean attributes not observerd by default?

### ISSUE THREAD

**Issue by pinguxx** [WebReflection/hyperHTML-Element#36]
Maybe its a bad understanding of my part but i would assume that by adding an attribute to booleanAttributes will also be observed but thats not the case, i have to put them in both places to make it work
https://stackblitz.com/edit/intermediate-ce-05?file=index.js
from the code above, if i do 
```
static get observedAttributes() {return ['attr1', 'active']; }
```
then the remove `els[1].removeAttribute('active');` works fine otherwise its not updated

Maybe make them be observed by default?

**Comment by WebReflection** [comment]
so ... you want booleans then you use `setAttribute` ? if you `els[1].active = false` you'll have observed behavior through `attributeCHangedCallback`, otherwise you can define whatever you want through all options.

Boolean attributes are meant to be used like in `<button disabled=${true || false} />` or `button.disabled = true || false`

**Comment by WebReflection** [comment]
also, just to be sure we are on the same page, to make boolean attributes observable you simply add the name in `observedAttributes` meaning it's more flexible but also more semantic.

To have what you need automatically you can simply write:

```js
static get booleanAttributes() { return ['active']; }
static get observedAttributes() {
  return this.booleanAttributes.concat('attr1');
}
```

That would work better with inheritance too.

TL;DR the idea of keeping boolean a part from magic behavior was meant, which is why those are defined before observedAttributes so that there are no interferences.

In this way all possibilities are open and putting a behavior together isn't difficult at all.

The most common use case for booleans in hyperHTML/Element is to define them right away while observing has a cost and if I can avoid implicitly adding a cost for all booleans used simply to style components or communicate a state, I'm happy doing that way.

I hope it's clear why it's like this, and I'll close this for now and I'll change eventually if convinced there is a need to penalize all boolean attributes by default.

**Comment by WebReflection** [comment]
**After thoughts** ... is it the `render()` that you are missing in there ? now that might be indeed handy to trigger on boolean changes, would that work? Still I wouldn't make booleans observable by default.

**Comment by WebReflection** [comment]
Actually you know what? I think your expectations should be matched.

**Comment by WebReflection** [comment]
Sorry for the monologue-storming ... I've fixed it, if you update dependencies in your example everything should work as expected.

Please let me know if that's not the case.

**Comment by pinguxx** [comment]
seems to be working, thanks for the fix, i think this is much better, no worries about the monologue i get to see your thought process on things, it's actually very helpful
